### PR TITLE
[[ TRAFODION 1815 ]]

### PR DESCRIPTION
**Pull request by amandamoran**

### CODE REVIEW

File: install/installer/traf_add_sudoAccess [hunk 1]
```diff
@@ -24,17 +24,28 @@ TRAF_CONFIG=/etc/trafodion/trafodion_config
...
+
+echo "## Allow $BACKUP_USER id to run commands needed for backup and restore" >> $BACKUP_USER
+echo "%$BACKUP_USER ALL =(hbase) NOPASSWD: /usr/bin/hbase" >> $BACKUP_USER
+
+}
 
 
+function copyTrafFile {
+
 if [[ $all_node_count -gt "1" ]]; then
```
Comment (DaveBirdsall) [inline install/installer/traf_add_sudoAccess]: No need for a $all_node_count -n test here?

File: install/installer/traf_config_setup [hunk 2]
```diff
@@ -555,7 +567,7 @@ echo "export DCS_PRIMARY_MASTER_NODE=\"$DCS_PRIMARY_MASTER_NODE\"" >> $LOCAL_TRA
 #==============================================
 #Enable HA
 
-echo -n "Enable High Availability (Y/N), default is N: "
+echo -n "Enable High Avalability (Y/N), default is N: "
```
Comment (DaveBirdsall) [inline install/installer/traf_config_setup]: The old code was correct here; the correct spelling is "Availability"

**Comment by DaveBirdsall** [comment]
Looks good to me. Though it would be good to fix the spelling error.

File: install/installer/traf_config_setup (hunk 2 above)
Comment (amandamoran) [inline install/installer/traf_config_setup]: Will do

File: install/installer/traf_add_sudoAccess (hunk 1 above)
Comment (amandamoran) [inline install/installer/traf_add_sudoAccess]: No it doesnt

**Comment by arvind-narain** [comment]
Sorry missed on the review yesterday.

1) Do please check on ENABLE_HA ("Y" vs "true") in traf_add_sudoaccess

2) Should there be an entry in trafodion_config_default for BACKUP_USER ?
Else should a -z or -n test be done before usage of BACKUP_USER ? Regression tests use
custom config file and since this var is not set the following errors are seen.
Not sure if just sourcing the file in after setting the var will solve both these issues or if sudo error is related to BACKUP_USER or something else.


***ERROR: BACKUP_USER variable not set in config file.
***WARNING: BACKUP_USER will be set to trafodion.
usage: sudo [-D level] -h | -K | -k | -V
usage: sudo -v [-AknS] [-D level] [-g groupname|#gid] [-p prompt] [-u user
            name|#uid]
usage: sudo -l[l] [-AknS] [-D level] [-g groupname|#gid] [-p prompt] [-U user
            name] [-u user name|#uid] [-g groupname|#gid] [command]
usage: sudo [-AbEHknPS] [-r role] [-t type] [-C fd] [-D level] [-g
            groupname|#gid] [-p prompt] [-u user name|#uid] [-g groupname|#gid]
            [VAR=value] [-i|-s] [<command>]
usage: sudo -e [-AknS] [-r role] [-t type] [-C fd] [-D level] [-g
            groupname|#gid] [-p prompt] [-u user name|#uid] file ...

......


***INFO: Creating trafodion sudo access file

/home/jenkins/workspace/core-regress-core-hdp/traf_inst/installer/traf_add_sudoAccess: line 41: $BACKUP_USER: ambiguous redirect
/home/jenkins/workspace/core-regress-core-hdp/traf_inst/installer/traf_add_sudoAccess: line 42: $BACKUP_USER: ambiguous redirect
cp: missing destination file operand after `/etc/sudoers.d'
Try `cp --help' for more information.

**Comment by amandamoran** [comment]
Thanks Arvind. I will take a look at this more carefully.

On Fri, Feb 19, 2016 at 10:46 AM, arvind-narain <notifications@github.com>
wrote:

> Sorry missed on the review yesterday.
>
> 1) Do please check on ENABLE_HA ("Y" vs "true") in traf_add_sudoaccess
>
> 2) Should there be an entry in trafodion_config_default for BACKUP_USER ?
> Else should a -z or -n test be done before usage of BACKUP_USER ?
> Regression tests use
> custom config file and since this var is not set the following errors are
> seen.
> Not sure if just sourcing the file in after setting the var will solve
> both these issues or if sudo error is related to BACKUP_USER or something
> else.
>
>
> ***ERROR: BACKUP_USER variable not set in config file. ***WARNING:
> BACKUP_USER will be set to trafodion.
> usage: sudo [-D level] -h | -K | -k | -V
> usage: sudo -v [-AknS] [-D level] [-g groupname|#gid] [-p prompt] [-u user
> name|#uid]
> usage: sudo -l[l] [-AknS] [-D level] [-g groupname|#gid] [-p prompt] [-U
> user
> name] [-u user name|#uid] [-g groupname|#gid] [command]
> usage: sudo [-AbEHknPS] [-r role] [-t type] [-C fd] [-D level] [-g
> groupname|#gid] [-p prompt] [-u user name|#uid] [-g groupname|#gid]
> [VAR=value] [-i|-s] []
> usage: sudo -e [-AknS] [-r role] [-t type] [-C fd] [-D level] [-g
> groupname|#gid] [-p prompt] [-u user name|#uid] file ...
>
> ......
>
> ***INFO: Creating trafodion sudo access file
>
> /home/jenkins/workspace/core-regress-core-hdp/traf_inst/installer/traf_add_sudoAccess:
> line 41: $BACKUP_USER: ambiguous redirect
> /home/jenkins/workspace/core-regress-core-hdp/traf_inst/installer/traf_add_sudoAccess:
> line 42: $BACKUP_USER: ambiguous redirect
> cp: missing destination file operand after /etc/sudoers.d'
> Trycp --help' for more information.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/328#issuecomment-186357078>
> .
>



-- 
Thanks,

Amanda Moran

**Pull request merged**